manim/mobject/types/vectorized_mobject.py

```diff
@@ -977,10 +1014,49 @@ def scale_handle_to_anchor_distances(self, factor: float):
         return self
 
     #
-    def consider_points_equals(self, p0, p1):
-        return np.allclose(p0, p1, atol=self.tolerance_for_point_equality)
+    def consider_points_equals(
+        self,
+        p0: Iterable[float] | Iterable[Iterable[float]],
+        p1: Iterable[float] | Iterable[Iterable[float]],
+    ) -> bool | np.ndarray:
+        """Determine if two points are close enough to be considered equal.
+
+        This function reimplements np.allclose, because repeated calling of
+        np.allclose for only 2 points is inefficient.
+        ----------
+        p0
+            first point
+        p1
+            second point
+
+        Returns
+        -------
+        bool | np.ndarray
+            Whether the points p0 and p1 are considered close or not.
+        """
+        p0 = np.asarray(p0)
+        p1 = np.asarray(p1)
+        atol = self.tolerance_for_point_equality
 
-    def consider_points_equals_2d(self, p0: np.ndarray, p1: np.ndarray) -> bool:
+        # Case 1: single pair of points
+        if p0.ndim == 1 and p1.ndim == 1:
+            if abs(p0[0] - p1[0]) > atol:
+                return False
+            if abs(p0[1] - p1[1]) > atol:
+                return False
+            if abs(p0[2] - p1[2]) > atol:
+                return False
+            return True
+
+        # Case 2: multiple pairs of points
+        is_close = abs(p1 - p0) <= atol
+        return is_close[:, 0] & is_close[:, 1] & is_close[:, 2]
```

Comment:
I've found that using methods like `np.all` or `np.sum` are actually slower when we only have to reduce 3 coordinates, so that's why I implemented this manually instead of doing something like `np.all(is_close, axis=1)` or `is_close.all(axis=1)`.

Reply:
Maybe I could add a comment in the code explaining this, though.

Reply:
Added some brief comments in the code regarding this.

Reply:
When you say "slower", how much slower are we actually talking?

Reply:
Well, using the same test scene:
- By using manual ANDs, `VMobject.consider_points_equals` takes 0.56% of the time, `VMobject.consider_points_equals_2d` has 1.13%, and in general `VMobject.get_subpath_split_indices_from_ponits` has 3.23%. 
![imagen](https://github.com/ManimCommunity/manim/assets/49853152/db5c6a68-09de-4791-92cb-5f57819bb7c6)

- By using `np.all`, `VMobject.consider_points_equals` takes 1.11% of the time, `VMobject.consider_points_equals_2d` has 1.76%, and in general `VMobject.get_subpath_split_indices_from_ponits` has 4.47%. 
![imagen](https://github.com/ManimCommunity/manim/assets/49853152/b2deaea3-fa64-40ca-8550-f0acd59602de)

And in general, by using `timeit` in a Jupyter Notebook I found out that using `np.all` takes twice as much time as using manual ANDs.

So, `VMobject.consider_points_equals` is 2x faster, `VMobject.consider_points_equals_2d` is around 1.5x faster, and `VMobject.get_subpath_split_indices_from_points` is around 1.25x faster.

So these three functions are considerably faster by themselves, although in the grand scale of things it doesn't influence too much in the render time of the whole scene.